### PR TITLE
feat: add create endpoints

### PR DESCRIPTION
**Pull request by gabriela-garcia-jalafund**
## Trello Ticket

[ID-02](https://trello.com/c/mmWvyxjq/22-id-02-crud-for-income-management)

---

## What did I do?

In this pull request, I implemented several endpoints for handling income records. This includes:

- A `POST /api/income` endpoint to allow users to log a new income, including details like the amount, source, and date.
- A `GET /api/income` endpoint to retrieve all logged incomes for the authenticated user.
- A `GET /api/income/{id}` endpoint to fetch a single income by its ID.
- A `PUT /api/income/{id}` endpoint to update an existing income.
- A `DELETE /api/income/{id}` endpoint to delete an income by its ID.
- Added input validation, specifically ensuring the amount is positive and the date format is valid (`yyyy-mm-dd`).

All database interactions are handled using Dapper, and the code is structured to ensure proper validation, error handling, and response formatting.

## Why did I do this?

The primary reason for these changes is to provide users with the ability to log, view, modify, and delete income records for financial tracking and reporting. This functionality is critical for allowing users to manage their personal finances effectively within the system. It also ensures that the system can accurately record and retrieve income data, offering better insights into users' financial activities.

## How did I do it?

I created several new API endpoints (`POST`, `GET`, `PUT`, and `DELETE`) for handling income records. The steps involved were:

1. Implementing the database operations using Dapper to ensure efficient queries and updates to the PostgreSQL database.
2. Adding input validation to ensure that the amount is a positive number and the date is in the correct format (`yyyy-mm-dd`).
3. Writing the logic to tie income entries to the correct user (`UserId`).
4. Creating appropriate response messages for success, validation errors, and authorization failures.
5. Ensuring that all operations are tested using tools like Postman and Swagger for proper functionality and error handling.

## (Optional) Notes:

- Review the input validation closely during the code review to ensure all edge cases are covered.
- Consider adding unit tests for each endpoint to verify the behavior with both valid and invalid data.

### CODE REVIEW

**Comment by gabriela-garcia-jalafund** [comment]
[ID-02: CRUD for Income Management](https://trello.com/c/mmWvyxjq/22-id-02-crud-for-income-management)

File: ExpenseTrackerGrupo2.csproj [hunk 1]
```diff
@@ -8,9 +8,17 @@
 
   <ItemGroup>
     <PackageReference Include="Dapper" Version="2.1.35" />
+    <PackageReference Include="DotNetEnv" Version="3.1.1" />
```
Comment (joseluis-teran-jalafund) [inline ExpenseTrackerGrupo2.csproj]: May you tell me why are you adding some new dependencies to the project?

File: src/ExpenseTrackerGrupo2/Controllers/IncomeController.cs [hunk 2]
```diff
@@ -0,0 +1,143 @@
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Threading.Tasks;
+using ExpenseTrackerGrupo2.Business.Services.Interfaces;
+using ExpenseTrackerGrupo2.Business.Services.Mappers.Requests.Incomes;
+
+namespace ExpenseTrackerGrupo2.API.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
```
Comment (joseluis-teran-jalafund) [inline src/ExpenseTrackerGrupo2/Controllers/IncomeController.cs]: I suggest to change our path to api/v1/income/[controller]

**Review by joseluis-teran-jalafund — CHANGES_REQUESTED** [review]

File: ExpenseTrackerGrupo2.csproj [hunk 3]
```diff
@@ -8,9 +8,17 @@
...
     <PackageReference Include="Microsoft.AspNetCore.OpenApi" Version="8.0.6" />
+    <PackageReference Include="Microsoft.Extensions.Configuration" Version="9.0.0-rc.1.24431.7" />
```
Comment (alex-paca-jalafund) [inline ExpenseTrackerGrupo2.csproj]: Mmm.... there is a lot of new tools  on csproj, maybe one or two it can be understandable, but there its actually more than 5, why its needed all the new tools?

**Review by alex-paca-jalafund — CHANGES_REQUESTED** [review]

File: ExpenseTrackerGrupo2.csproj (hunk 1 above)
Comment (gabriela-garcia-jalafund) [reply ExpenseTrackerGrupo2.csproj]: I was trying another solution, for what alex already did, but I will delete it

File: src/ExpenseTrackerGrupo2/Controllers/IncomeController.cs (hunk 2 above)
Comment (gabriela-garcia-jalafund) [reply src/ExpenseTrackerGrupo2/Controllers/IncomeController.cs]: Done.

File: ExpenseTrackerGrupo2.csproj (hunk 3 above)
Comment (gabriela-garcia-jalafund) [reply ExpenseTrackerGrupo2.csproj]: I was trying another solution, for what you already did and I forgot to delete it, but I already deleted it.

File: src/ExpenseTrackerGrupo2/Controllers/IncomeController.cs [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+            if (request.Amount <= 0)
+            {
+                return BadRequest("Amount must be a positive number.");
+            }
+
+            try
+            {
+                var updatedIncome = await _incomeService.UpdateIncome(request);
+                if (updatedIncome == null)
```
Comment (alex-paca-jalafund) [inline src/ExpenseTrackerGrupo2/Controllers/IncomeController.cs:97]: Solve the warnings, i review the functionality and the implementation its working, but pls try to dont add warnnings.

**Review by alex-paca-jalafund — CHANGES_REQUESTED** [review]

**Review by alex-paca-jalafund — APPROVED** [review]

**Review by joseluis-teran-jalafund — APPROVED** [review]
